packages/pnpm-sync-lib/src/pnpmSyncPrepare.ts

```diff
@@ -229,3 +240,59 @@ function processDependencies(
     }
   }
 }
+
+// process all dependencies and devDependencies to find potential transitive injected dependencies
+// and add to injectedDependencyToFilePath map
+function processTransitiveInjectedDependency(
+  pnpmLockfile: ILockfile | undefined,
+  injectedDependencyToVersion: Map<string, Set<string>>
+): void {
+  const potentialTransitiveInjectedDependencyVersionQueue: Array<string> = [];
+  for (const injectedDependencyVersion of [...injectedDependencyToVersion.values()]) {
+    for (const version of [...injectedDependencyVersion]) {
+      potentialTransitiveInjectedDependencyVersionQueue.push(version);
+    }
+  }
+
+  const lockfilePackages: Record<string, ILockfilePackage> | undefined = pnpmLockfile?.packages;
+
+  if (lockfilePackages) {
+    while (potentialTransitiveInjectedDependencyVersionQueue.length > 0) {
+      const transitiveInjectedDependencyVersion: string | undefined =
+        potentialTransitiveInjectedDependencyVersionQueue.shift();
+      if (transitiveInjectedDependencyVersion) {
+        const { dependencies, optionalDependencies } = lockfilePackages[transitiveInjectedDependencyVersion];
+        processInjectedDependencies(
+          dependencies,
+          injectedDependencyToVersion,
+          potentialTransitiveInjectedDependencyVersionQueue
+        );
+        processInjectedDependencies(
+          optionalDependencies,
+          injectedDependencyToVersion,
+          potentialTransitiveInjectedDependencyVersionQueue
+        );
+      }
+    }
+  }
+}
+function processInjectedDependencies(
+  dependencies: Record<string, IVersionSpecifier> | undefined,
+  injectedDependencyToVersion: Map<string, Set<string>>,
+  potentialTransitiveInjectedDependencyVersionQueue: Array<string>
+): void {
+  if (dependencies) {
+    for (const [dependency, specifier] of Object.entries(dependencies)) {
+      const specifierToUse: string = typeof specifier === 'string' ? specifier : specifier.version;
```

Comment:
Is `specifier.version` actually possible here?  Seems like these entries always come from `pnpmLockfile?.packages`